### DISCUSSION TITLE
Bad request response in python SDK

### DISCUSSION THREAD

**Discussion by wildanvin** [bacalhau-project/bacalhau#1752] [Q&A] [answered]
Hello,
I've been trying the python sdk that is on branch `1481-generate-python-sdk-via-swagger`
The first 2 examples are working pretty good but when I run the last example `submit_job.py` I get a bad request error.
What I am doing is `python3 submit_job.py` and this is the error that I get (the last part):
```
 File "/home/wilman/.local/lib/python3.10/site-packages/bacalhau_apiclient/rest.py", line 228, in request
    raise ApiException(http_resp=r)
bacalhau_apiclient.rest.ApiException: (400)
Reason: Bad Request
HTTP response headers: HTTPHeaderDict({'Content-Type': 'text/plain; charset=utf-8', 'Ratelimit-Limit': '1000', 'Ratelimit-Remaining': '999', 'Ratelimit-Reset': '1', 'X-Bacalhau-Client-Id': '4b1b7a1f6f0117b8d98b5b07a501f033a909f283e05dcff4fc267a29dcd173b2', 'X-Content-Type-Options': 'nosniff', 'X-Rate-Limit-Duration': '1', 'X-Rate-Limit-Limit': '1000.00', 'X-Rate-Limit-Request-Remote-Addr': '157.100.137.71:43352', 'Date': 'Mon, 23 Jan 2023 17:47:16 GMT', 'Content-Length': '195'})
HTTP response body: {"Code":"error-unknown-server-error","Message":"client's signature is invalid: crypto/rsa: verification error","Details":{},"Err":"client's signature is invalid: crypto/rsa: verification error"}
```

It says that the client's signature is invalid, maybe I need to update a crypto library? but it is weird because when I submit a job from the CLI it works perfectly.

Thanks in advance for any insights.

**Comment by enricorotundo** [comment, marked as answer]
Hey @wildanvin thanks for bringing this up! The `1481-generate-python-sdk-via-swagger` branch changes the way the client's signature is validated and unfortunately that's not been merged to `main` yet so it's not even deployed to production yet. 
I promise will merge within this week 😃 , and then the python SDK will be usable with the next release (ie `v0.3.18`).

Having said that, If you're really eager to give the SDK a try 🔥 , you can spin up a [local bacalhau devstack](https://docs.bacalhau.org/running-node/test-network/) and set the `BACALHAU_API_HOST` and `BACALHAU_API_PORT` env variables in the terminal you use to run `submit_job.py`. The SDK will [pick them up](https://github.com/filecoin-project/bacalhau/blob/7b33bcc862bc78f1af28894da23bd72547fd8f60/python/bacalhau_sdk/config.py#L61) and use the local bacalhau instance.

**Reply by wildanvin** [reply]
Ohh I see, I will wait for the next release then... will try to spin up a local node too.
Thanks a lot for your answer and the helpful links :v:

**Reply by enricorotundo** [reply]
Feel free to reach me out on Slack if you need help :)
